test/test_eos_stiff.cpp

```diff
@@ -0,0 +1,303 @@
+      auto h_pressure = Kokkos::create_mirror_view(v_pressure);
+      auto h_bulkmodulus = Kokkos::create_mirror_view(v_bulkmodulus);
+      auto h_gruneisen = Kokkos::create_mirror_view(v_gruneisen);
+#else
+      // Create arrays for the outputs and then pointers to those arrays that
+      // will be passed to the functions in place of the Kokkos views
+      std::array<Real, num> h_temperature;
+      std::array<Real, num> h_pressure;
+      std::array<Real, num> h_bulkmodulus;
+      std::array<Real, num> h_gruneisen;
+      // Just alias the existing pointers
+      auto v_temperature = h_temperature.data();
+      auto v_pressure = h_pressure.data();
+      auto v_bulkmodulus = h_bulkmodulus.data();
+      auto v_gruneisen = h_gruneisen.data();
+#endif // PORTABILITY_STRATEGY_KOKKOS
+
+      WHEN("A P(rho, e) lookup is performed") {
+        eos.PressureFromDensityInternalEnergy(v_density, v_energy, v_pressure, num);
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::fence();
+        Kokkos::deep_copy(h_pressure, v_pressure);
+#endif // PORTABILITY_STRATEGY_KOKKOS
+        THEN("The returned P(rho, e) should be equal to the true value") {
+          array_compare(num, density, energy, h_pressure, pressure_true, "Density",
+                        "Energy");
+        }
+      }
+
+      WHEN("A B_S(rho, e) lookup is performed") {
+        eos.BulkModulusFromDensityInternalEnergy(v_density, v_energy, v_bulkmodulus, num);
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::fence();
+        Kokkos::deep_copy(h_bulkmodulus, v_bulkmodulus);
+#endif // PORTABILITY_STRATEGY_KOKKOS
+        THEN("The returned B_S(rho, e) should be equal to the true value") {
+          array_compare(num, density, energy, h_bulkmodulus, bulkmodulus_true, "Density",
+                        "Energy");
+        }
+      }
+
+      WHEN("A T(rho, e) lookup is performed") {
+        eos.TemperatureFromDensityInternalEnergy(v_density, v_energy, v_temperature, num);
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::fence();
+        Kokkos::deep_copy(h_temperature, v_temperature);
+#endif // PORTABILITY_STRATEGY_KOKKOS
+        THEN("The returned B_S(rho, e) should be equal to the true value") {
+          array_compare(num, density, energy, h_temperature, temperature_true, "Density",
+                        "Energy");
+        }
+      }
+
+      WHEN("A Gamma(rho, e) lookup is performed") {
+        eos.GruneisenParamFromDensityInternalEnergy(v_density, v_energy, v_gruneisen,
+                                                    num);
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::fence();
+        Kokkos::deep_copy(h_gruneisen, v_gruneisen);
+#endif // PORTABILITY_STRATEGY_KOKKOS
+        THEN("The returned Gamma(rho, e) should be equal to the true value") {
+          array_compare(num, density, energy, h_gruneisen, gruneisen_true, "Density",
+                        "Energy");
+        }
+      }
+    }
+  }
+}
```

Comment:
@c0sm0-kramer can you add the following tests:

- [ ] Test against a finite difference approximation of the bulk modulus (i.e. the same formula but use pressure lookups to calculate the finite differences).
- [ ] Make sure that $S(\rho, e) == S(\rho, T(\rho, e)$, i.e. for a given energy and density test that `TemperatureFromDensityInternalEnergy()` followed by `EntropyFromDensityTemperature()` gives the same result as `EntropyFromDensityInternalEnergy()`.
- [ ] Test that you can set the `qq`, `Pinf`, and `qp` parameters equal to zero and recover the ideal gas behavior for at least some of the functions (as many as you're willing).

Reply:
@jhp-lanl tests now added